spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/Span.java

```diff
@@ -305,7 +305,7 @@ public void logEvent(String event) {
 	 * Will never be null.
 	 */
 	public Map<String, String> tags() {
-		return Collections.unmodifiableMap(this.tags);
+		return Collections.unmodifiableMap(new LinkedHashMap<>(this.tags));
```

Comment:
if the goal is thread safety, you might need to synchronize on tags, else you could get a CME on the copy-out